### PR TITLE
Add option status in crab tasks command. Fixes #4526.

### LINKED ISSUE #4526: Add option to filter tasks by status in crab tasks command

### PR DESCRIPTION
**Pull request by nizamyusli**

### CODE REVIEW

File: src/python/CRABClient/Commands/tasks.py [hunk 1]
```diff
@@ -16,10 +16,21 @@ def __call__(self):
         dictresult.sort()
         dictresult.reverse()
 
+        if self.options.status:
+            dictresult = [item for item in dictresult if item[1] == self.options.status]
+
         today = date.today() 
 
+        if dictresult == []:
```
Comment (AndresTanasijczuk) [inline src/python/CRABClient/Commands/tasks.py]: I would change this to "if not dictresult:"

File: src/python/CRABClient/Commands/tasks.py [hunk 2]
```diff
@@ -67,3 +84,8 @@ def validateOptions(self):
                 self.date = date.today() - timedelta(days=days)
             except ValueError:
                 raise ValueError("\nPlease enter a NUMBER\nExample: crab tasks --days=10")
+        if self.options.status:
+            ## TODO: this shouldn't be hard-coded.
+            if self.options.status not in ['NEW', 'HOLDING', 'QUEUED', 'UPLOADED', 'SUBMITTED', 'RESUBMIT', 'FAILED', 'UNKNOWN']:
+                self.logger.info("\nPlease enter a valid STATUS\nExample: NEW, UPLOADED, SUBMITTED, FAILED\n")
+                sys.exit()
```
Comment (AndresTanasijczuk) [inline src/python/CRABClient/Commands/tasks.py]: In other commands we use raise ConfigurationException(msg) when validation of an option fails. Can we do the same here?

File: src/python/CRABClient/Commands/tasks.py [hunk 3]
```diff
@@ -16,10 +16,21 @@ def __call__(self):
...
+        if self.options.status:
+            dictresult = [item for item in dictresult if item[1] == self.options.status]
+
         today = date.today() 
 
+        if dictresult == []:
+            self.logger.info ('No tasks from %s until %s' %(self.date, today))
+            if self.options.status:
+                self.logger.info ('with the STATUS %s' %(self.options.status))
+            sys.exit()
```
Comment (AndresTanasijczuk) [inline src/python/CRABClient/Commands/tasks.py]: Why sys.exit() and not return?

File: src/python/CRABClient/Commands/tasks.py [hunk 4]
```diff
@@ -16,10 +16,21 @@ def __call__(self):
...
 
+        if self.options.status:
+            dictresult = [item for item in dictresult if item[1] == self.options.status]
+
         today = date.today() 
 
+        if dictresult == []:
+            self.logger.info ('No tasks from %s until %s' %(self.date, today))
+            if self.options.status:
+                self.logger.info ('with the STATUS %s' %(self.options.status))
```
Comment (AndresTanasijczuk) [inline src/python/CRABClient/Commands/tasks.py]: "with the STATUS" -->  "with status"

File: src/python/CRABClient/Commands/tasks.py [hunk 5]
```diff
@@ -16,10 +16,21 @@ def __call__(self):
...
+            sys.exit()
+
         self.logger.info ('\n')
         self.logger.info ('The list of tasks from %s until %s' %(self.date, today))
+        if self.options.status:
+            self.logger.info ('with the STATUS %s' %(self.options.status))
```
Comment (AndresTanasijczuk) [inline src/python/CRABClient/Commands/tasks.py]: "with the STATUS" -->  "with status"

**Comment by AndresTanasijczuk** [comment]
@nizamyusli On top of the comments I made, can you also remove the white spaces between "self.logger.info" and "("?

**Comment by nizamyusli** [comment]
Got it.

**Pull request merged**